### PR TITLE
feature: create basic auth

### PR DESCRIPTION
**Pull request by DariuszMech**

### CODE REVIEW

File: src/main/java/org/example/auth/JwtAuthenticator.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import io.dropwizard.auth.Authenticator;
+import io.jsonwebtoken.Jwts;
+import org.example.models.JwtToken;
+import org.example.models.UserRole;
+
+import java.security.Key;
+import java.util.Optional;
+
+public class JwtAuthenticator implements Authenticator<String, JwtToken> {
+    Key key;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/auth/JwtAuthenticator.java:13]: private

File: src/main/java/org/example/controllers/AuthController.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                                key = HttpHeaders.AUTHORIZATION,
+                                name = HttpHeaders.AUTHORIZATION,
+                                in = ApiKeyAuthDefinition.ApiKeyLocation.HEADER
+                        )
+                }
+        )
+)
+public class AuthController {
+
+    AuthService authService;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/controllers/AuthController.java:39]: private

File: src/main/java/org/example/models/JwtToken.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package org.example.models;
+
+import javax.security.auth.Subject;
+import java.security.Principal;
+
+public class JwtToken implements Principal {
+
+    UserRole userRole;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/JwtToken.java:8]: private

File: src/main/java/org/example/models/LoginRequest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package org.example.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class LoginRequest {
+
+    String username;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/LoginRequest.java:8]: privates

File: src/main/java/org/example/models/User.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package org.example.models;
+
+public class User {
+    final String username;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/User.java:4]: privates

**Review by amordzon — APPROVED** [review]
super

**Pull request merged**